flang/lib/Parser/prescan.cpp

```diff
@@ -997,20 +997,27 @@ const char *Prescanner::FixedFormContinuationLine(bool mightNeedSpace) {
   }
   tabInCurrentLine_ = false;
   char col1{*nextLine_};
-  if (InCompilerDirective()) {
-    // Must be a continued compiler directive.
-    if (!IsFixedFormCommentChar(col1)) {
-      return nullptr;
-    }
+  if (InCompilerDirective() ||
+      (features_.IsEnabled(LanguageFeature::OpenMP) &&
+          IsFixedFormCommentChar(col1) && nextLine_[1] == '$')) {
     int j{1};
-    for (; j < 5; ++j) {
-      char ch{directiveSentinel_[j - 1]};
-      if (ch == '\0') {
-        break;
-      }
-      if (ch != ToLowerCaseLetter(nextLine_[j])) {
+    if (InCompilerDirective()) {
```

Comment:
Restructure this `if` statement nest as `if (InCompilerDirective()) { ... } else if (features_.InEnabled(...)) { }`.  It's needlessly confusing now as it stands.